### PR TITLE
CMake: Use Imported Targets for 3rd Party Libs

### PR DESCRIPTION
**Pull request by sgiraudot**
## Summary of Changes

Following [this discussion](https://github.com/CGAL/cgal/pull/4624), this is a PR to switch from the function-based 3rd party handling (`CGAL_target_use_[lib]()`) to the imported targets framework (`target_link_libraries([myprogram] CGAL::CGAL CGAL::[lib]_support)`).

## Release Management

* Affected package(s): all (cmake)

### CODE REVIEW

**Comment by lrineau** [comment]
That looks good, so far. The `CMakeLists.txt` is complex, but it was complex before this set of changes anyway.

**Comment by sgiraudot** [comment]
Alright, I did a full pass on the whole library, all `target_use` should be replaced by imported targets. I updated the doc of 3rd party libraries accordingly.

I completely removed all the `CGAL_target_use_...` functions as they were never in a release (no need to deprecate them). The only exception is `CGAL_target_use_TBB` for which I added a deprecation message.

**Comment by lrineau** [comment]
I think this PR should also change the sentence we find now in the developer manual:
https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-5.1-Ic-127/output2/Manual/devman_create_and_use_a_cmakelist.html#seclink
> There are also some cmake macros to link with CGAL dependencies that can be found in the section Essential and Optional Third Party Dependencies.

It was not macros, but functions, and anyway now it will become imported targets.

**Comment by lrineau** [comment]
By the way, do we want this PR in CGAL-5.1? I think so.

**Comment by sloriot** [comment]
Side remark, I noticed that for basic viewer some extra flags must be passed. I wonder whether a similar macro shouldn't be defined . CC @gdamiand

**Comment by gdamiand** [comment]
> Side remark, I noticed that for basic viewer some extra flags must be passed. I wonder whether a similar macro shouldn't be defined . CC @gdamiand

Only add_definitions(-DCGAL_USE_BASIC_VIEWER); not sure it deserves an imported target ?

Maybe we can define this flag automatically when cgal_qt5 is defined ? (and thus automatically enable basic viewer when cgal_qt5 is found) ?

**Comment by sgiraudot** [comment]
> By the way, do we want this PR in CGAL-5.1? I think so.

I strongly agree: the functions `CGAL_target_use...` were never in a release, integrating this PR directly will save users the trouble of changing their cmake scripts twice.

**Comment by lrineau** [comment]
@sgiraudot There are Travis errors to fix before it can be tested.

**Comment by maxGimeno** [comment]
errors : 
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.1-Ic-150/Polyhedron_Demo/TestReport_afabri_x64_Cygwin-Windows10_MSVC2015-Debug-64bits.gz

and I think 
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.1-Ic-150/Polyhedron_Demo/TestReport_lrineau_ArchLinux-clang.gz
also

**Comment by maxGimeno** [comment]
Some hints might be found here : https://github.com/PointCloudLibrary/pcl/pull/3341/files

**Comment by maxGimeno** [comment]
Conflicts

**Comment by lrineau** [comment]
@sgiraudot @maxGimeno In my opinion, this PR is one that we want in 5.1 (a blocker). You should fix the conflicts as soon as possible, and test the PR again in `integration`.

**Comment by sgiraudot** [comment]
I'm on it.

**Comment by sgiraudot** [comment]
Rebased.

**Comment by maxGimeno** [comment]
Thank you !

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.1-Ic-168.shtml

File: Installation/cmake/modules/CGAL_Eigen_support.cmake [hunk 1]
```diff
@@ -0,0 +1,9 @@
+if(EIGEN3_FOUND AND NOT TARGET CGAL::Eigen_support)
+  if(NOT TARGET Threads::Threads)
+    find_package(Threads REQUIRED)
+  endif()
```
Comment (lrineau) [inline Installation/cmake/modules/CGAL_Eigen_support.cmake:4]: @sgiraudot If you are still around, do you remember why you search for CMake `Threads` support in a file about Eigen3? Was that a copy-paste error?

**Pull request merged**